### PR TITLE
[breaking] Motion JPEG scheduling

### PR DESCRIPTION
**Pull request by steabert**
Instead of playing JPEG images at network speed, schedule them according to their RTP timestamps.
This introduces a clock (to determine playback time) and changes to the canvas component.
Additionally, all the examples have been reworked to better deal with the multiple different setups
we can have now.

### CODE REVIEW

File: examples/browser/README.md [hunk 1]
```diff
@@ -1,22 +1,70 @@
...
+Or, to use the camera example directly, browse directly to
+`http://localhost:8080/camera/simple.html`.
+
+## Camera example
+
+After serving the examples with `yarn examples` and browsing
+to a file under `http://localhost:8080/camera`, you will need to enter
+the camera (device) IP address and choose an encoding.
+After that, just click the `play` button and you should
+see the live video appearing.
```
Comment (Tigge) [inline examples/browser/README.md]: "see the live video appear."

File: examples/browser/README.md [hunk 2]
```diff
@@ -1,22 +1,70 @@
...
+The examples should work with any Axis camera, although
+it should be easy to be modified for any other server as
```
Comment (Tigge) [inline examples/browser/README.md]: "it should be easily modified"

File: examples/browser/README.md [hunk 3]
```diff
@@ -1,22 +1,70 @@
...
+long as it supports proxying RTSP/RTP over WebSocket.
+
+The example `simple.html` shows how you can include `media-stream-library.min.js`
+directly in your html file to then use to create a player.
+The corresponding `simple-player.js` file implements a minimal player,
+which can be a starting point to create a more advanced player.
```
Comment (Tigge) [inline examples/browser/README.md]: "which can be used as a starting point for creating a more advanced player."

File: examples/browser/README.md [hunk 4]
```diff
@@ -1,22 +1,70 @@
...
 
-You have to make sure you've built the `media-stream-library.min.js` bundle
-by running `yarn && yarn build` in the top directory.
-Then, just serve the contents of this directory:
+The examples under the `test` subdirectory all use a test server,
+which has to be launched first by running `yarn rtsp`.
+It's best to do this in a separate terminal window, so that
+you can keep it running.
 
+Either use no special arguments to launch the RTSP servers:
```
Comment (Tigge) [inline examples/browser/README.md]: Or do what? What is the other option when we specify arguments?

File: examples/browser/README.md [hunk 5]
```diff
@@ -1,22 +1,70 @@
...
 ```
-yarn http-server
+yarn rtsp
 ```
 
-### Create your own player
+You can even add your own launch pipeline, for more details see the
```
Comment (Tigge) [inline examples/browser/README.md]: "[...] pipeline. For more details see the"

File: examples/browser/camera/simple-player.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+const {pipelines} = window.mediaStreamLibrary
+
+// force auth
+const authorize = async (host) => {
+  // Force a login by fetching usergroup
+  const fetchOptions = {
+    credentials: 'include',
+    headers: {
+      'Axis-Orig-Sw': true,
+      'X-Requested-With': 'XMLHttpRequest'
```
Comment (Tigge) [inline examples/browser/camera/simple-player.js]: is the "X-Requested-With" header necessary?

File: examples/browser/test/h264-overlay-player.js [hunk 7]
```diff
@@ -50,4 +50,4 @@ const play = (host) => {
 }
 
 // Each time a device ip is entered, authorize and then play
```
Comment (Tigge) [inline examples/browser/test/h264-overlay-player.js]: Is it doing auth still?

File: examples/browser/test/h264-player.js [hunk 8]
```diff
@@ -17,4 +17,4 @@ const play = (host) => {
 }
 
 // Each time a device ip is entered, authorize and then play
```
Comment (Tigge) [inline examples/browser/test/h264-player.js]: Is it doing auth still?

File: examples/browser/test/mjpeg-player.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    rtsp: { uri: `rtsp://localhost:8555/test` },
+    mediaElement
+  })
+  pipeline.ready.then(() => {
+    console.log('pipeline ready')
+    pipeline.play()
+  })
+}
+
+// Each time a device ip is entered, authorize and then play
```
Comment (Tigge) [inline examples/browser/test/mjpeg-player.js]: Is it doing auth still?

File: rtsp-ws-server/start.sh [hunk 10]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# This script launches an RTSP server (using docker) and
```
Comment (Tigge) [inline rtsp-ws-server/start.sh]: Wooo, indentation slope coming up!

File: lib/utils/clock.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      this.started = window.performance.now()
+      this.stopped = false
+    }
+  }
+
+  stop () {
+    if (!this.stopped) {
+      const now = window.performance.now()
+      this.elapsed = this.elapsed + (now - this.started)
+      this.started = now
```
Comment (Tigge) [inline lib/utils/clock.js]: I this needed?

Comment (steabert) [reply lib/utils/clock.js]: no, should be removed

**Pull request merged**